pkg/gitopsjob/job.go

```diff
@@ -0,0 +1,240 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitopsjob
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var gitopsJobImage = "quay.io/redhat-appstudio/gitops-generator:latest"
```

Comment:
nit: I'd declare this as a const

Reply:
Yeah, oops. Agreed.

Reply:
Fixed.

Reply:
Is it wise to pull in the latest?  what if there are breaking changes from the repo?  We should probably consider having releases on the gitops-generator repo and pulling in image versions that we know are compatible/tested with HAS eventually

Reply:
We need to start doing releases in gitops-generator repo.. But if we dont pull in the latest here, we are going to get some of the errors back we face in infra-deploy PR 1291!

Reply:
yes, agreed for now but something to think about once we stabilize